Make script element IDL attributes reflect content attributes

vakila opened the pull request:
Per #253, change the spec to make the .event and .htmlFor IDL attributes
of the HTMLScriptElement reflect the "event" and "for" content attributes.

The corresponding test must also be changed (see w3c/web-platform-tests#2280).

zcorpan commented on source:
```diff
@@ -111888,8 +111888,11 @@ if (s = prompt('What is your name?')) {
-  <p>The <dfn><code data-x="dom-script-event">event</code></dfn> and <dfn><code data-x="dom-script-htmlFor">htmlFor</code></dfn> IDL attributes of the <code>script</code>
-  element must return the empty string on getting, and do nothing on setting.</p>
+  <p>The <dfn><code data-x="dom-script-event">event</code></dfn> IDL attribute of the 
+  <code>script</code> element must reflect the element's event content attribute.</p>
```
s/event/<code data-x="attr-script-event">event</code>

similarly for `for` below

zcorpan commented on source:
```diff
@@ -111888,8 +111888,11 @@ if (s = prompt('What is your name?')) {
+  <code>script</code> element must reflect the element's event content attribute.</p>
+
+  <p>The <dfn><code data-x="dom-script-htmlFor">htmlFor</code></dfn> IDL attributes of the 
+  <code>script</code> element must must reflect the element's for content attribute.</p>
```
two musts

zcorpan commented on source:
```diff
@@ -111888,8 +111888,11 @@ if (s = prompt('What is your name?')) {
+  <p>The <dfn><code data-x="dom-script-event">event</code></dfn> IDL attribute of the 
+  <code>script</code> element must reflect the element's event content attribute.</p>
+
+  <p>The <dfn><code data-x="dom-script-htmlFor">htmlFor</code></dfn> IDL attributes of the 
```
s/attributes/attribute/

zcorpan commented:
Can you prepend the commit message with "Fixes #253:" ?

vakila commented:
@zcorpan Thanks very much for reviewing this and catching those problems. I've corrected the double "must" and changed the commit message as directed, and also fixed another typo and added some missing definition links. But I'm afraid I don't understand what's meant by "s/event/event" and  "s/attributes/attribute/". Could you please explain?

zcorpan commented:
s/foo/bar/ means replace foo with bar. :-) (looks like github interpreted my markup for the "event" comment above...)

LGTM

zcorpan commented:
Thank you @vakila! Committed as 6c635b1ca71ab6b12d1e3ad2438457778554eabd.

(There was some trailing spaces that I removed -- maybe you can change your editor to trim trailing whitespace, for future PRs?)

vakila commented:
Thanks for the explanation @zcorpan! :) And yes, I'll set my editor to remove trailing whitespace in the future.
